lib/msprime.c

```diff
@@ -4951,8 +4951,14 @@ msp_run_sweep(msp_t *self)
 
     ret = model->params.sweep.generate_trajectory(
         &model->params.sweep, self, &num_steps, &time, &allele_frequency);
+
+    /* Solution 1 to rare case of zero waiting time until first event
+     * is to shift times by nextafter up front */
+    /* t_start = nextafter(self->time, DBL_MAX); */
+
     t_start = self->time;
     sweep_dt = model->params.sweep.trajectory_params.genic_selection_trajectory.dt;
+    tsk_bug_assert(sweep_dt > 0);
```

Comment:
putting asserts on `dt > 0` won't get at the issue, right? you need to check that whatever you add `dt` to is actually changed by doing so.

Reply:
that's right-- this is pure sanity checking. just want to show you guys that it's not the issue